geode-wan/src/main/java/org/apache/geode/cache/client/internal/locator/wan/LocatorHelper.java

```diff
@@ -46,13 +46,42 @@ public static boolean addLocator(int distributedSystemId, DistributionLocatorId
     Set<DistributionLocatorId> existingValue =
         allLocatorsInfo.putIfAbsent(distributedSystemId, locatorsSet);
     if (existingValue != null) {
-      if (!existingValue.contains(locator)) {
+      if (locator.getMemberName() != null) {
+        DistributionLocatorId tempLocator = null;
+        for (DistributionLocatorId locElement : existingValue) {
```

Comment:
This code is pretty much duplicated below.  It might read better if you extracted it to a method like getLocatorWithSameMemberName(locatorId, locatorIdCollection)